src/lib/vocab/community/communityServices.ts

```diff
@@ -234,6 +243,11 @@ export const LeaveCommunityService: ServiceByName['LeaveCommunity'] = {
 			// 	repos.persona.findById(persona_id),
 			// 	repos.community.findById(community_id),
 			// ]);
+
+			if (actor !== persona_id || !(await checkCommunityAccess(actor, community_id, repos))) {
+				return {ok: false, status: 403, message: 'actor does not have permission'};
+			}
```

Comment:
Two bugs here:

- `checkCommunityAccess` needs to be unwrapped, this would always evaluate to false (I checked, we don't make this mistake elsewhere)
- `checkCommunityAccess` should always be checked, not conditionally, but on the `persona_id` not the `actor`
- `isPersonaAdmin` should be checked

```suggestion
			if (actor !== persona_id && !(await isPersonaAdmin(actor, repos))) {
				return {ok: false, status: 403, message: 'actor does not have permission'};
			}
			unwrap(await checkCommunityAccess(persona_id, community_id, repos));
```

Reply:
Good catch, thanks!